File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n=nums.size();
        for(auto& i:nums){
            if(nums[abs(i)]<0) return abs(i);
            nums[abs(i)]=-nums[abs(i)];
        }
        return -1;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        # for each val -> idx -> -(nums[val])
        # for duplicates it already set that value to -(nums[val]) 
        # that would be the result 

        for val in nums: 
            idx = abs(val)
            if nums[idx]<0: 
                return idx
            else: 
                nums[idx]=-nums[idx]
        return -1